ansible_collections/arista/avd/plugins/action/eos_designs_facts.py

```diff
@@ -86,9 +96,9 @@ def run(self, tmp=None, task_vars=None):
 
         return result
 
-    def create_avd_switch_facts_instances(self, fabric_hosts: list, hostvars: object, default_vars: dict):
+    def get_all_hostvars(self, fabric_hosts, hostvars: object, default_vars: dict):
```

Comment:
```suggestion
    def get_all_hostvars(self, fabric_hosts: list, hostvars: object, default_vars: dict):
```

Reply:
So here - looking at the usage of the `failed` value in the `run` funciton we could actually break out of the loop as soon as one host fails, it seems we don't need the complete all_hostvars as we will return immediately with empty result anyway.

This could optimise failure detection for big topologies.

Reply:
This would stop validation of input data from all other hosts, so we would have to rerun again and again for every fix. This way we get all validation issues at once.

Reply:
hmmm we are not returning anything pertaining to validation right? we just look for `result['failed'] and we don't do anything with the result messages in the dictionary:
https://github.com/aristanetworks/ansible-avd/blob/devel/ansible_collections/arista/avd/plugins/plugin_utils/schema/avdschematools.py#L96

Though depending on the input validation mode the messages _could_ be printed in ansible Display indeed but not in every case

Reply:
The messages would be printed according to validation_mode setting, which is `warning` by default.